lib/api/model/model.dart

```diff
@@ -698,3 +710,83 @@ class DmMessage extends Message {
   @override
   Map<String, dynamic> toJson() => _$DmMessageToJson(this);
 }
+
+enum MessageEditState {
+  none,
+  edited,
+  moved;
+
+  // Adapted from the shared code:
+  //   https://github.com/zulip/zulip/blob/1fac99733/web/shared/src/resolved_topic.ts
+  // Pattern for an arbitrary resolved-topic prefix.
+  // These always begin with the canonical prefix, but can go on longer.
+  // It's designed to remove a weird "✔ ✔✔ " prefix, if present.
+  static RegExp _resolvedTopicPrefixRe = RegExp('^✔ [ ✔]*');
+
+  static bool _isResovled(String topic) {
+    return _resolvedTopicPrefixRe.matchAsPrefix(topic) != null;
```

Comment:
This has the right behavior, I think, but can be simplified:
```suggestion
    return topic.startsWith('✔ ');
```

but probably that prefix should actually be pulled out as a `static const` member on the class so that it gets a name.

That simplification makes it a bit easier to reason through — no thinking about regexps — and should also be more efficient.